windowing-system/windowing-system.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable max-classes-per-file */
+// @ts-check
+
+/**
+ * Implement the classes etc. that are needed to solve the
+ * exercise in this file. Do not forget to export the entities
+ * you defined so they are available for the tests.
+ */
+export class Size {
+  constructor(width = 80, height = 60) {
+    this.width = width;
+    this.height = height;
+  }
+
+  resize(newWidth, newHeight) {
+    this.width = newWidth;
+    this.height = newHeight;
+  }
+}
+
+export class Position {
+  constructor(x = 0, y = 0) {
+    this.x = x;
+    this.y = y;
+  }
+
+  move(newX, newY) {
+    this.x = newX;
+    this.y = newY;
+  }
+}
+
+export class ProgramWindow {
+  constructor() {
+    this.screenSize = new Size(800, 600);
+    this.size = new Size();
+    this.position = new Position();
+  }
+
+  resize(newSize) {
+    if (newSize.width < 1) {
+      newSize.width = 1;
```

Comment:
Mutating the `newSize` object argument is not encouraged. Let's think of a different approach that doesn't imply mutating its value. Additionally, you can simplify the amount of `if` statements by using the `Math.min` and `Math.max` functions.